src/main/scala/com/fulcrumgenomics/bam/ErrorRateByReadPosition.scala

```diff
@@ -197,48 +201,62 @@ class ErrorRateByReadPosition
   def readNumber(rec: SAMRecord): Int = if (!rec.getReadPairedFlag) 0 else if (rec.getFirstOfPairFlag) 1 else 2
 }
 
-private class ObsCounter(readNumber: Int, position: Int) {
-  var a_ref_obs: Long  = 0
-  var c_ref_obs: Long  = 0
-  var a_to_c_obs: Long = 0
-  var a_to_g_obs: Long = 0
-  var a_to_t_obs: Long = 0
-  var c_to_a_obs: Long = 0
-  var c_to_g_obs: Long = 0
-  var c_to_t_obs: Long = 0
-
-  /** Increments the appropriate counter. */
-  def count(ref: Char, read: Char): Unit = (ref, read) match {
-    case ('A', 'A') => a_ref_obs  += 1
-    case ('A', 'C') => a_to_c_obs += 1
-    case ('A', 'G') => a_to_g_obs += 1
-    case ('A', 'T') => a_to_t_obs += 1
-    case ('C', 'A') => c_to_a_obs += 1
-    case ('C', 'C') => c_ref_obs  += 1
-    case ('C', 'G') => c_to_g_obs += 1
-    case ('C', 'T') => c_to_t_obs += 1
-    case _          => unreachable("Should never be invoked with a case other than the above.")
+/** Counter for substitutions for a given read number and position in that read.
+  *
+  * @param readNumber the read number
+  * @param position the position in the read
+  * @param collapse true to collapse substitution types, false otherwise
+  */
+private class ObsCounter(readNumber: Int, position: Int, collapse: Boolean) extends SimpleCounter[(Char, Char)] {
```

Comment:
I figured this would be slower, so I ran a little test on a sample BAM that I have kicking around.  The runtime of the metric collection phase (ignoring starting) grew by 50%.  I think this is all due to using SimpleCounter.  I have a suggestion on how to fix and will make a PR as I think that's easiest.

Reply:
Merged your PR!